src/main/java/de/qabel/desktop/config/DefaultClientConfiguration.java

```diff
@@ -70,6 +80,37 @@ public void selectIdentity(Identity identity) {
 	}
 
 	@Override
+	public Date getLastDropPoll(Identity identity) {
+
+		Date dropDate = lastDropMap.get(identity.getKeyIdentifier());
+		if (dropDate != null) {
+			return dropDate;
+		}
+		Date firstDropDate = new Date(0L);
+		setLastDropPoll(identity, firstDropDate);
+		return firstDropDate;
```

Comment:
`hint:`
default pattern for this method would be has-put-get:
```JAVA
String key = identity.getKeyIdentifier();
if (!lastDropMap.containsKey(key)) {
  lastDropMap.put(key, new Date(0L);
}
retrun lastDropMap.get(key);
```

the observer notification done by `setLastDropPoll` caused by a call to `getLastDropPoll` is also superfluous.